python/ray/data/tests/test_mongo_dataset.py

```diff
@@ -0,0 +1,261 @@
+import pandas as pd
+import pyarrow as pa
+import pytest
+
+import ray
+from ray.data.datasource import MongoDatasource
+
+from ray.data.tests.conftest import *  # noqa
+from ray.data.tests.mock_http_server import *  # noqa
+from ray.tests.conftest import *  # noqa
+
+
+@pytest.fixture
+def start_mongo():
+    import subprocess
+
+    subprocess.run(["sudo", "apt-get", "purge", "-y", "mongodb*"])
+    subprocess.run(["sudo", "apt-get", "install", "-y", "mongodb"])
```

Comment:
this won't work on mac. I am thinking we should add an integration test / nightly test later.

Reply:
Other os can have a different setup. Also this will likely move out to ci env setup.